codice/ddf
# DDF-6408 Improve Logging for CDM

#### What does this PR do?
Improves logging in the **catalog-core-directorymonitor** bundle:
  - CDM now has its own log in **data/logs**
  - Concurrent set keeps track of files still being processed in **AsyncFileAlterationObserver**
  - Additional Logging information in cdm classes

#### Who is reviewing it? 
@mcalcote 
@aaronilovici 

#### Select relevant component teams: 
@codice/core-apis 

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@paouelle

#### How should this be tested?
1. Pull and build this branch
2. Set up DDF with a normal profile and set cdmLogging to TRACE
3. Create a directory of files to use for the cdm. There should be 20 or more files in this directory. Duplicate all of the files to a subdirectory within the first directory.
    3a. Add permissions for the directory in **security/configurations.policy**. Message me for details 
4. Set up a "Monitor in place" CDM in the admin console by going to **Catalog -> Configuration -> Catalog Content Directory Monitor**, be sure to specify the path of the directory created in step 3
5. Open cdm.log in **data/log**. Verify that the CDM was successfully set up. Verify that CDM shows "files processing", the number of files that need to be committed to the monitor. Also make sure that the logs occasionally display a list of files still needing to be processed.

#### What are the relevant tickets?
Fixes: #6408

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
